components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/hosting/LinkManagerHostingMapImpl.kt

```diff
@@ -115,26 +117,40 @@ internal class LinkManagerHostingMapImpl(
         }
     }
 
-    private fun HostedIdentityEntry.toGroupIdWithPublicKeyHash(): GroupIdWithPublicKeyHash {
-        val publicKey = publicKeyReader.loadPublicKey(this.sessionPublicKey)
-        return GroupIdWithPublicKeyHash(
-            this.holdingIdentity.groupId,
-            ByteBuffer.wrap(keyHasher.hash(publicKey)),
-        )
+    private fun HostedIdentityEntry.toGroupIdWithPublicKeyHash(): Collection<GroupIdWithPublicKeyHash> {
+        return this.alternativeSessionKeys.map {
+            it.toGroupIdWithPublicKeyHash(this.holdingIdentity.groupId)
+        } + this.prefferedSessionKey.toGroupIdWithPublicKeyHash(this.holdingIdentity.groupId)
     }
 
     private fun addEntry(entry: HostedIdentityEntry) {
         val info = HostingMapListener.IdentityInfo(
             holdingIdentity = entry.holdingIdentity.toCorda(),
             tlsCertificates = entry.tlsCertificates,
             tlsTenantId = entry.tlsTenantId,
-            sessionPublicKey = publicKeyReader.loadPublicKey(entry.sessionPublicKey),
-            sessionCertificates = entry.sessionCertificates
+            preferredSessionKey = entry.prefferedSessionKey.toData(),
         )
         locallyHostedIdentityToIdentityInfo[entry.holdingIdentity.toCorda()] = info
-        publicHashToIdentityInfo[entry.toGroupIdWithPublicKeyHash()] = info
+        entry.toGroupIdWithPublicKeyHash().forEach {
+            publicHashToIdentityInfo[it] = info
+        }
         listeners.forEach {
             it.identityAdded(info)
         }
     }
+
+    private fun HostedIdentitySessionKey.toData(): HostingMapListener.SessionKey {
```

Comment:
```suggestion
    private fun HostedIdentitySessionKey.toData(): HostingMapListener.SessionKey {
```
```suggestion
    private fun HostedIdentitySessionKey.toCorda(): HostingMapListener.SessionKey {
```
minor, but for consistency with other converter functions.